graspy/match/faq.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2019 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import math
+from scipy.optimize import linear_sum_assignment
+from scipy.optimize import minimize_scalar
+from .skp import SinkhornKnopp
+
+
+class FastApproximateQAP:
+    """
+    Fast Approximate QAP Algorithm (FAQ)
+    The FAQ algorithm solves the Quadratic Assignment Problem (QAP) finding an alignment of the
+    vertices of two graphs which minimizes the number of induced edge disagreements [1].
+    
+    
+    Parameters
+    ----------
+    
+    n_init : int
+        Number of random initializations of the starting permutation matrix that the FAQ algorithm will undergo.
+        n_init automatically set to 1 if init_method = 'barycenter'
+        
+    init_method : string
+        The initial position chosen
+        "barycenter" : the non-informative “flat doubly stochastic matrix,” J=1*1^T/n, i.e the barycenter of
+        the feasible region
+        "rand" : some random point near J, (J+K)/2, where K is some random doubly stochastic matrix
+        
+    Attributes
+    ----------
+    
+    perm_inds_ : array, size (n,1) where n is the number of vertices in the graphs fitted
+        The indices of the optimal permutation found via FAQ
```

Comment:
can you explain what this is a little bit more? Most important thing is, which matrix is being permuted? is it the case that if I call fit on `(A, B)` then `A` should match `B[np.ix_(perm_inds, perm_inds)]`? If so, make sure to specify which thing it being permuted, and say that this is permuting the order of nodes in the adjacency matrix